x-pack/plugins/observability_solution/dataset_quality/common/data_stream_details/types.ts

```diff
@@ -19,3 +17,13 @@ export interface UpdateFieldLimitParams {
   dataStream: string;
   newFieldLimit: number;
 }
+
+export interface CheckAndLoadIntegrationParams {
+  dataStream: string;
+}
+
+export interface IntegrationType {
+  isIntegration: boolean;
+  areAssetsAvailable: boolean;
+  integration?: Integration;
+}
```

Comment:
**doubt**: is there a direct correlation between this type and the rt definition [here](https://github.com/elastic/kibana/pull/198692/files#diff-d6e6732e8f65c1856abbb34628a002d63665261c513b1bbe58291c25f7e74ccfR100)? Or is there some misalignment and this type is not used with the rt type response?

Reply:
There is a correlation in that the API response contains this data structure in an identical shape. But in the interest of not coupling the types used on the data quality page with the API types I think it makes sense to define them separately. If one of them changes the compiler would point out the point where a conversion happens.